### PR TITLE
Added token calculation using tiktoken for gpt 4 vision and code update for latest version 1.3.5

### PR DESCRIPTION
**Pull request by RamishRasool14**
## Summary

Briefly describe the changes and the goal of this PR. Make sure the PR title summarizes the changes effectively.
* Added token calculation using tiktoken for gpt 4 vision for both image urls and base64 images
* Modified code to support latest version of openai 1.3.5
## Motivation

Why are these changes necessary? How do they improve the cookbook?
* Previously there was no support for input image prompt token calculation
* The code was using older version of openai which was incorrect when executed using latest version
---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [x] I have added a new entry in [registry.yaml](/registry.yaml) (and, optionally, in [authors.yaml](/authors.yaml)) so that my content renders on the cookbook website.
- [x] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [x] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [x] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [x] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [x] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [x] Correctness: The information I include is correct and all of my code executes successfully.
  - [x] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: registry.yaml [hunk 1]
```diff
@@ -4,6 +4,15 @@
 # should build pages for, and indicates metadata such as tags, creation date and
 # authors for each page.
 
+- title: GPT-4 vision image token calculation using TikToken
```
Comment (simonpfish) [inline registry.yaml:7]: We already have an entry for `examples/How_to_count_tokens_with_tiktoken.ipynb`

If you want this to be a new entry, we should split it into its own notebook.

File: examples/How_to_count_tokens_with_tiktoken.ipynb [hunk 2]
```diff
@@ -433,7 +434,84 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Tokens: 1445. Reference https://platform.openai.com/docs/guides/vision/calculating-costs\n"
+     ]
+    }
+   ],
+   "source": [
+    "import math\n",
```
Comment (simonpfish) [inline examples/How_to_count_tokens_with_tiktoken.ipynb:449]: Could be nice to add a text cell above explaining what's happening here and citing the docs.

**Review by simonpfish — CHANGES_REQUESTED** [review]
This is awesome, thanks! 

A couple of small comments.

**Comment by dongmingli-Ben** [comment]
Several new models came out in the last month, including gpt-4-1106-preview. It would be nice to update the token counting example to include these new models.

**Pull request closed without merging**